src/components/Navbar/Navbar.js

```diff
@@ -11,59 +12,42 @@ import { ThemeContext } from '../../context/ThemeContex';
 import { switchLanguage } from '../../i18n/switchLanguage';
 import { SidebarMenu } from '../SidebarMenu/SidebarMenu';
 import {
-  darkActiveLanguageButton,
-  darkInactiveLanguageButton,
   desktopMenu,
   desktopMenuDark,
   desktopMenuItem,
+  languageButton,
+  languageButtonDark,
   languageThemeWraper,
-  lightActiveLanguageButton,
-  lightInactiveLanguageButton,
-  line,
-  lineDark,
   logo,
   logoContainer,
   mainContainer,
   mainContainerDark,
   name,
   sidebar,
   sidebarContainer,
+  themeButton,
 } from './Navbar.module.scss';
 
 export function Navbar() {
   const [isSidebarMenuVisible, setIsSidebarMenuVisible] = useState(false);
-  const [language, setLanguage] = useState('pl');
-  const [buttonColorPL, setButtonColorPL] = useState('');
-  const [buttonColorENG, setButtonColorENG] = useState('');
   const { isDarkTheme, toggleTheme } = useContext(ThemeContext);
   const { theme } = useContext(ThemeContext);
   const { t } = useTranslation();
+  const [currentLanguage, setCurrentLanguage] = useState('pl');
 
   function toggleSidebarMenu() {
     setIsSidebarMenuVisible((prev) => !prev);
   }
 
-  const handleButtonChange = (lang) => {
-    setLanguage(lang);
-  };
-
-  useEffect(() => {
-    if (language === 'pl') {
-      setButtonColorPL(
-        theme(lightActiveLanguageButton, darkActiveLanguageButton)
-      );
-      setButtonColorENG(
-        theme(lightInactiveLanguageButton, darkInactiveLanguageButton)
-      );
-    } else if (language === 'en') {
-      setButtonColorENG(
-        theme(lightActiveLanguageButton, darkActiveLanguageButton)
-      );
-      setButtonColorPL(
-        theme(lightInactiveLanguageButton, darkInactiveLanguageButton)
-      );
+  const handleButtonChange = () => {
```

Comment:
Tę funkcję można wziąć z hook `useCallback`. I jako wyzwalacze dodać `switchLanguage` oraz `currentLanguage` aby funkcja poprawnie działała. 
Przykład: 
```
const handleButtonChange = useCallback(() => {
    if (currentLanguage === 'pl') {
      switchLanguage('en');
      setCurrentLanguage('en');
    } else {
      switchLanguage('pl');
      setCurrentLanguage('pl');
    }
}, [switchLanguage, currentLanguage])
```
Poprawi to wydajność